common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -127,7 +127,7 @@
   },
   "BINARY_ARITHMETIC_OVERFLOW" : {
     "message" : [
-      "<value1> <symbol> <value2> caused overflow."
+      "<value1> <symbol> <value2> caused overflow.<alternative>"
```

Comment:
Please, add two sub-classes: one with hit and another without, see for instance:
https://github.com/apache/spark/blob/036db74b97f8f5b447bc1d689e9f8081af47604c/common/utils/src/main/resources/error/error-conditions.json#L3692-L3709